e2e/test/iothub/properties/PropertiesE2ETests.cs

```diff
@@ -307,7 +307,20 @@ await Task
                 .ConfigureAwait(false);
 
             // Validate the updated properties from the device-client
-            ClientProperties clientProperties = await deviceClient.GetClientPropertiesAsync().ConfigureAwait(false);
+
+            // Since this will depend on the SDK updating the properties asynchronously under-the-cover,
+            // we will need to fetch the properties until we receive an updated version number.
```

Comment:
The `UpdateClientPropertiesAysnc` call now happens asynchronously on a different thread, internally. As a result, when the `writablePropertyCallbackSemaphore` is released, we won't have updated the property yet. The customer will now need to loop over the `DeviceClient.GetClientPropertiesAsync()` call and validate against the `Version` to compare and check if that has been updated.